modules/exploits/windows/misc/weblogic_deserialize.rb

```diff
@@ -48,10 +56,48 @@ def initialize(info={})
       'DisclosureDate' => 'Apr 17 2018'))
   end
 
+  def check
+    connect
+    req = "GET /console/login/LoginForm.jsp HTTP/1.1\n"
+    req << "Host: #{peer}\n\n"
+    sock.put(req)
+
+    res = sock.get_once
+    disconnect
+    return CheckCode::Unknown unless res
+
+    /WebLogic Server Version: (?<version>\d+\.\d+\.\d+\.*\d*)/ =~ res
+    if version
+      version = Gem::Version.new(version)
```

Comment:
I'm thinking we're gonna want something less boilerplatey for version comparison. This is fine for now, but we really need library code for this.

Reply:
(I'll do it. :)